src/NuGet.Core/NuGet.Protocol/Plugins/PluginCacheEntry.cs

```diff
@@ -0,0 +1,110 @@
+            NewCacheFileName = CacheFileName + "-new";
+        }
+
+        internal TimeSpan MaxAge { get; set; } = TimeSpan.FromDays(30);
+        internal string CacheFileName { get; }
+        private string RootFolder { get; }
+        private string NewCacheFileName { get; }
+
+        public IReadOnlyList<OperationClaim> OperationClaims { get; private set; }
+
+        public void LoadFromFile()
+        {
+            Stream content = null;
+            try
+            {
+                content = CachingUtility.TryReadCacheFile(MaxAge, CacheFileName);
+                if (content != null)
+                {
+                    ProcessContent(content);
+                }
+            }
+            catch
+            {
+                content?.Dispose();
+            }
+            finally
+            {
+                content?.Dispose();
+            }
+        }
+
+        private void ProcessContent(Stream content)
+        {
+            var serializer = new JsonSerializer();
+            using (var sr = new StreamReader(content))
+            using (var jsonTextReader = new JsonTextReader(sr))
+            {
+                OperationClaims = serializer.Deserialize<IReadOnlyList<OperationClaim>>(jsonTextReader);
+            }
+        }
+
+        public void AddOrUpdateOperationClaims(IReadOnlyList<OperationClaim> operationClaims)
+        {
+            OperationClaims = operationClaims;
+        }
+
+        public async Task UpdateCacheFileAsync()
+        {
+            if (OperationClaims != null)
+            {
+                // Make sure the cache file directory is created before writing a file to it.
+                DirectoryUtility.CreateSharedDirectory(RootFolder);
+
+                // The update of a cached file is divided into two steps:
+                // 1) Delete the old file.
+                // 2) Create a new file with the same name.
+                using (var fileStream = new FileStream(
+                    NewCacheFileName,
+                    FileMode.Create,
+                    FileAccess.ReadWrite,
+                    FileShare.None,
+                    CachingUtility.BufferSize,
+                    useAsync: true))
+                {
+                    var json = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(OperationClaims, Formatting.Indented));
+                    await fileStream.WriteAsync(json, 0, json.Length);
+                }
+
+                if (File.Exists(CacheFileName))
+                {
+                    // Process B can perform deletion on an opened file if the file is opened by process A
+                    // with FileShare.Delete flag. However, the file won't be actually deleted until A close it.
+                    // This special feature can cause race condition, so we never delete an opened file.
+                    if (!CachingUtility.IsFileAlreadyOpen(CacheFileName))
+                    {
+                        File.Delete(CacheFileName);
```

Comment:
FileUtility.Delete or DeleteWithLock

Reply:
FileUtility.Delete and DeleteWithLock could take up to 300ms 
Consistency is not as important here to take the extra perf hit. 
We do the same thing in our http caching.

Reply:
ACtually it slipped my mind...https://github.com/NuGet/NuGet.Client/pull/2299/files#diff-f2833de1ecaed93179251b95dde8d2a5R210

There is already a cache file lock here. 

There shouldn't be any consistency issues here.